config/pmd-test.xml

```diff
@@ -10,139 +10,111 @@
+        <!-- This rule does the same as EmptyCatchBlock, and has a confusing message -->
         <exclude name="JUnitUseExpected"/>
-        <!--will be addressed in https://github.com/checkstyle/checkstyle/projects/5-->
-        <exclude name="NonThreadSafeSingleton"/>
-    </rule>
-
-    <rule ref="rulesets/java/codesize.xml/NcssMethodCount">
-        <properties>
-            <!-- 'testTokenNumbering' method is simple but long as it recheck each token one by one -->
-            <!-- 'testNospaceWithComplexInput' it is intended to keep all in one method. -->
-            <!--Method `testTokenValues` in JavadocTokenTypes contains several asserts as it checks each token explicitly-->
-            <!-- XdocsPagesTest is a complicated list of different checks -->
-            <property name="violationSuppressXPath" value="
-            //ClassOrInterfaceDeclaration[@Image='GeneratedJavaTokenTypesTest']//MethodDeclarator[@Image='testTokenNumbering']
-            | //ClassOrInterfaceDeclaration[@Image='ParenPadCheckTest']//MethodDeclarator[@Image='testNospaceWithComplexInput']
-            | //MethodDeclaration[@Name='testTokenValues' and ../../..[@Image='JavadocTokenTypesTest']]
-            | //MethodDeclaration[@Name='getModulePropertyExpectedValue' and ../../..[@Image='XdocsPagesTest']]"/>
-        </properties>
-    </rule>
-
-    <rule ref="rulesets/java/codesize.xml/ExcessiveMethodLength">
-        <properties>
-            <!-- 'testTokenNumbering' method is simple but long as it recheck each token one by one -->
-            <!-- 'testNospaceWithComplexInput' it is intended to keep all in one method. -->
-            <!--Method `testTokenValues` in JavadocTokenTypes contains several asserts as it checks each token explicitly-->
-            <!-- XdocsPagesTest is a complicated list of different checks -->
-            <property name="violationSuppressXPath" value="
-            //ClassOrInterfaceDeclaration[@Image='GeneratedJavaTokenTypesTest']//MethodDeclarator[@Image='testTokenNumbering']
-            | //ClassOrInterfaceDeclaration[@Image='ParenPadCheckTest']//MethodDeclarator[@Image='testNospaceWithComplexInput']
-            | //ClassOrInterfaceDeclaration[@Image='ParenPadTest']//MethodDeclarator[@Image='testMethodParen']
-            | //MethodDeclaration[@Name='testTokenValues' and ../../..[@Image='JavadocTokenTypesTest']]
-            | //MethodDeclaration[@Name='getModulePropertyExpectedTypeName' and ../../..[@Image='XdocsPagesTest']]"/>
-        </properties>
+        <!-- We do not want invest extra time to make logic ideal in test method, may be later -->
+        <exclude name="NPathComplexity"/>
+        <!-- We do throws Exception, as any exception that in not caught in test should fail a test
+            that is expected behavior and we like it as it follow fail-fast and helps to avoid extra
+            catch blocks -->
+        <exclude name="SignatureDeclareThrowsException"/>
```

Comment:
what was a reason to move this exclude from previous position ?

Right now it is very complicated to verify diff as too much is reallocated.
If possible please minimize diff.
if there is some good reason in new order, please let me know, but it will be very hard to validate your changes.

Reply:
They moved all the checks. I had to do this file from scratch.

Reply:
Please do us a huge favor, please try to minimize diff, to speed review.

Reply:
Believe me, I did my best. The new rules are so messed that it is not possible to keep the order.
They did not just replace `ruleset` with `category`, they regrouped all the rules.

Reply:
> Believe me, 

whole process in checkstyle, whole this madness with numerous CIs/tools/.... is based on  "I do not trust contributors, they always cheat, ....". I had a lot of stories on how I trusted contributors and in result they left incomplete/damaged functionality. Most contributions are done to start use new functionality after release. Non of contributor should be able to block others to upgrade to new version after release.
My time on laptop is very limited, so process is designed to make me do most of work from phone and on the go, or when I have only 5 min of free time, ... .

But I do appreciate your help in project.